source/MRMesh/MRColorMapAggregator.cpp

```diff
@@ -0,0 +1,161 @@
+#include "MRColorMapAggregator.h"
+#include "MRGTest.h"
+
+namespace MR
+{
+
+template<typename Tag>
+void ColorMapAggregator<Tag>::setDefaultColor( const Color& color )
+{
+    defaultColor_ = color;
+}
+
+template<typename Tag>
+void ColorMapAggregator<Tag>::pushBack( const ColorMap& colorMap, const ElementBitSet& elementsBitSet )
+{
+    checkInputData_( colorMap, elementsBitSet );
+    dataSet_.push_back( { colorMap, elementsBitSet } );
+}
+
+template<typename Tag>
+void ColorMapAggregator<Tag>::insert( int i, const ColorMap& colorMap, const ElementBitSet& elementsBitSet )
+{
+    assert( i < dataSet_.size() );
+    checkInputData_( colorMap, elementsBitSet );
+    dataSet_.push_back( { colorMap, elementsBitSet } );
```

Comment:
why push_back, `i` can be less than `dataSet_.size()`

Reply:
fixed